[Diagram] Board 이벤트 다이어그램 작성

vwx12345 opened the pull request:
### 💻 작업 내용
- 게시판 생성 다이어그램 추가
- 게시판 삭제 다이어그램 추가
- 게시판 조회 다이어그램 추가

### ✨ 리뷰 포인트
- queryBoard.puml 에서 특정 user에 대해 조회할 때, memberId로 할 지 nickname으로 할지 고민이고 이 부분은 domain과 일관되게 작성해야할 것 같습니다.

### 📝 메모
- Domain 생성 시, 팩토리 패턴 도입 고민 중입니다.

### 🎯 관련 이슈
<!-- pr이 merge 되면 이슈가 자동으로 close 되도록 합니다. 만약 자동 close 를 하지 않고 이슈만 링크한다면 closed 키워드를 삭제해주세요.-->
closed #이슈번호

Commit:
docs(diagram): 게시글 생성 시나리오 시퀀스 다이어그램 추가

Commit:
docs(diagram): 게시글 삭제 시나리오 시퀀스 다이어그램 추가

Commit:
docs(diagram): 게시글 목록 조회 시나리오 시퀀스 다이어그램 추가

Commit:
Merge branch 'develop' of https://github.com/SimpleBoardTeam/simpleBoard-server into document/diagram-event-board/#5

vwx12345 committed:
Merge remote-tracking branch 'origin/develop' into document/diagram-event-board/#5

vwx12345 committed:
"chore(directory): auth context domain, member event 다이어그램 전체 구조 변경으로 삭제"

Commit:
infra(diagram): 중복된 directory 제거

Commit:
docs(diagram): title 생성

Commit:
docs(diagram): deleteBoard 상태 코드 기반 처리로 변경

Commit:
Merge remote-tracking branch 'origin/develop' into document/diagram-event-board/#5

vwx12345 committed:
chore(diagram): auto-generate SVGs

Commit:
docs(diagram): 권한 체크 API 로 변경

vwx12345 committed:
chore(diagram): auto-generate SVGs

codingbaraGo commented on document/puml/event/externalAPI/boardContext/boardAggregate/createBoard.puml:
```diff
@@ -0,0 +1,28 @@
+BoardCommandService -> DomainEventPublisher : publish(BoardCreatedEvent)
+
+BoardCommandService --> BoardController : BoardId
+BoardController --> Client : HTTP 201 + BoardId
```
이 반환 이후에 클라이언트(웹브라우저)의 화면은 어디로 이동 예정인가요?
게시판 생성 페이지가 따로 있나요?
만약 생성된 게시판의 메인페이지로 이동한다면 게시판 이름도 필요할 것 같네요

codingbaraGo replied:
Post 쪽에서 게시글 조회시 board name을 사용하기 때문에 board name도 같이 반환해줘야 할 것 같아요

codingbaraGo commented on document/puml/event/externalAPI/boardContext/boardAggregate/queryBoard.puml:
```diff
@@ -0,0 +1,24 @@
+participant BoardQueryService
+participant BoardQueryRepository
+
+User -> BoardController : GET /boards? \npage={page}&size={size}&searchType={type}&keyword={kw}&nickname={nickname}
+BoardController -> BoardQueryService : getBoards(SearchCondition searchCondition)
```
닉네임은 어디에 사용되나요?
API 명세상에는 검색 타입이 title만 보이네요
board 개설자 닉네임 검색 기능 같은게 있는건가요?

vwx12345 replied:
> 닉네임은 어디에 사용되나요? API 명세상에는 검색 타입이 title만 보이네요 board 개설자 닉네임 검색 기능 같은게 있는건가요?

 board 개설자 닉네임 검색 기능이 맞습니다. 그래서 필요하다고 판단해서 API 명세서에 반영해놓겠습니다.

codingbaraGo replied:
아래 코멘트에 대한 답변을 보니 개설자 닉네임 검색이 아니라 board 최고권한 매니저 닉네임 검색 기능이 아닐까 하는데
매니저 닉네임 검색 맞나요?

codingbaraGo commented on document/puml/event/externalAPI/boardContext/boardAggregate/queryBoard.puml:
```diff
@@ -0,0 +1,24 @@
+* page : int
+* size : int
+* title : String
+1. nickname : String (작성자 닉네임)
+2. memberId : Long (만약 이 방식이면 nickname으로 Member B.C. 에서 가져와야 함)
```
작성자 == 게시판 생성자 라고 생각하면 될까요?
아니면 작성자 == 게시판 최고권한 매니저?

vwx12345 replied:
이 부분은 서비스랑 연결되는 부분이라 상의해볼 필요가 있겠네요. 당장은 게시판 최고권한 매니저 입니다.

codingbaraGo replied:
저도 게시판 최고권한 매니저가 맞다고 생각합니다.
작성자라고 적힌 부분을 수정하기만 하면 될 것 같네요!

codingbaraGo commented on document/puml/event/externalAPI/boardContext/boardAggregate/deleteBoard.puml:
```diff
@@ -0,0 +1,31 @@
+
+actor User
+participant BoardController
+participant BoardService
+participant BoardRepository
```
다른 이벤트 다이어그램에는 CQRS가 적용되어있는데 여기에는 없네요.
통일하는게 나중에 보고 구현하기 좋을 것 같아요.

vwx12345 replied:
> 다른 이벤트 다이어그램에는 CQRS가 적용되어있는데 여기에는 없네요. 통일하는게 나중에 보고 구현하기 좋을 것 같아요.

맞는 말입니다. 반영하겠습니다.

codingbaraGo reviewed (commented):
고생하셨습니다!
메이저한 수정사항은 없어보이네요.

Commit:
docs(diagram): deleteBoard 에 CQRS 도입

Commit:
docs(diagram): 의미없는 dto 제거, BoardCreatedEvent 구체화
- boardName과 {boardName, userId} 만 가지는 dto 는 굳이 필요없단 판단하여 제거했습니다.
- BoardCreatedEvent 에 어떤 값이 들어가는 지 추가했습니다.

Commit:
docs(diagram): participant 위치 수정

Commit:
docs(diagram): DomainEventPublisher 색 추가

Commit:
docs(diagram): permission API true/false 기반으로 변경

Commit:
docs(diagram): Dto 작업

vwx12345 committed:
chore(diagram): auto-generate SVGs
